website/content/en/docs/building-operators/golang/webhook.md

```diff
@@ -0,0 +1,300 @@
+│   │   └── webhookcainjection_patch.yaml
+│   ├── manager
+│   │   ├── controller_manager_config.yaml
+│   │   ├── kustomization.yaml
+│   │   └── manager.yaml
+│   ├── manifests
+│   │   └── kustomization.yaml
+│   ├── prometheus
+│   │   ├── kustomization.yaml
+│   │   └── monitor.yaml
+│   ├── rbac
+│   │   ├── auth_proxy_client_clusterrole.yaml
+│   │   ├── auth_proxy_role.yaml
+│   │   ├── auth_proxy_role_binding.yaml
+│   │   ├── auth_proxy_service.yaml
+│   │   ├── kustomization.yaml
+│   │   ├── leader_election_role.yaml
+│   │   ├── leader_election_role_binding.yaml
+│   │   ├── memcached_editor_role.yaml
+│   │   ├── memcached_viewer_role.yaml
+│   │   ├── role_binding.yaml
+│   │   └── service_account.yaml
+│   ├── samples
+│   │   ├── cache_v1alpha1_memcached.yaml
+│   │   └── kustomization.yaml
+│   ├── scorecard
+│   │   ├── bases
+│   │   │   └── config.yaml
+│   │   ├── kustomization.yaml
+│   │   └── patches
+│   │       ├── basic.config.yaml
+│   │       └── olm.config.yaml
+│   └── webhook
+│       ├── kustomization.yaml
+│       ├── kustomizeconfig.yaml
+│       └── service.yaml
+├── controllers
+│   ├── memcached_controller.go
+│   └── suite_test.go
+├── go.mod
+├── go.sum
+├── hack
+│   └── boilerplate.go.txt
+└── main.go
+
+19 directories, 53 files
+```
+
+The scaffoled file `api/v1alpha1/memcached_webhook.go` has `ValidateCreate`, `ValidateUpdate`, and `ValidateDelete` functions, 
+which allows you to perform different validations based on the operation being performed. The mutating webhook implementation belongs in the `Default` function.
+
+The memcached operator we are building in this example is too simple to require defaulting or additional validations, 
+but as an example, we can reinforce that the default value for `spec.size` should be `3`, by adding the following logic to 
+the `Default` function (note that this is already handled by the CRD defaulting and is technically completely superfluous):
+
+```sh
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Memcached) Default() {
+	memcachedlog.Info("default", "name", r.Name)
+
+	if r.Spec.Size == 0 {
+		r.Spec.Size = 3
+	}
+}
+```
+
+For validation, we can enforce that the size of the cluster follows a rule that is difficult or impossible 
+to describe with OpenAPI, for example, that the size of the cluster always remain an odd number.
+
+In order to perform this validation, we can simply add below `validateOdd` function that will perform this check.
+
+```sh
+func validateOdd(n int32) error {
+	if n%2 == 0 {
+		return errors.New("Cluster size must be an odd number")
+	}
+	return nil
+}
+```
+
+At the end, call this method from `ValidateCreate` and `ValidateUpdate` function as shown below.
+
+```sh
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *Memcached) ValidateCreate() error {
+	memcachedlog.Info("validate create", "name", r.Name)
```

Comment:
```suggestion
log.Info("validate create", "name", r.Name)
```
lets have `log.Info` for now as we have not defined `memcachedlog` anywhere.

Reply:
Modified.